posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,506 @@
+
+=== More about transforming server configurations
+
+Transformation of a server configuration is only necessary when java EE package names are exposed in the server configuration.  One case of explosed java EE packages names are JMS activation specification elements.  For example:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="javax.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+This must be transformed to:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="jakarta.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+== Updating features in server configurations:
+
+When running a Jakarta EE 9 enable application with Open-Liberty, Jakarta EE 9 versions of Open-Liberty features must be specified in the server configuration.
+
+In many cases, Jakarta EE 9 features use the same short names as the corresponding Java EE features.  The Jakarta EE 9 features simply update the feature version.  Here is a complete table of (non-internal) features which were updated by changing just the feature version:
+
+.Java EE to Jakarta EE 9 Feature Updates: Version Only
+|===
+| Java EE 8 feature name | Jakarta EE 9 feature name
+
+| appClientSupport-1.0
+| appClientSupport-2.0
+
+| appSecurity-3.0
+| appSecurity-4.0
+
+| batch-1.0
+| batch-2.0
+
+| batchManagement-1.0
+| batchManagement-2.0
+
+| beanValidation-2.0
+| beanValidation-3.0
+
+| cdi-2.0
+| cdi-3.0
+
+| concurrent-1.0
+| concurrent-2.0
+
+| javaee-8.0
+| jakartaee-9.0
+
+| javaeeClient-8.0
+| jakartaeeClient-9.0
+
+| jsonb-1.0
+| jsonb-2.0
+
+| jsonbContainer-1.0
+| jsonbContainer-2.0
+
+| jsonp-1.1
+| jsonp-2.0
+
+| jsonpContainer-1.1
+| jsonpContainer-2.0
+
+| managedBeans-1.0
+| managedBeans-2.0
+
+| mdb-3.2
+| mdb-4.0
+
+| servlet-4.0
+| servlet-5.0
+
+| webProfile-8.0
+| webProfile-9.0
+
+| websocket-1.1
+| websocket-2.0
+|===
+
+In other cases, in addition to updating the feature version, the feature short name was updated.
+
+To simplify updating server configuration files, a Jakarta EE 9 feature name which uses the new feature version while retaining the old feature name is also supported.
+
+.Java EE to Jakarta EE 9 Feature Updates: Short Name and Version
+|===
+| Java EE 8 feature name | Jakarta EE 9 feature name | Alternative Jakarta EE 9 feature name
```

Comment:
Change `Java EE 8 feature name` to `Jakarta EE 9 feature name` and remove the `Alternatiive Jakarta EE 9 feature name` and of course remove the column from below.

Reply:
See above.